packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -886,18 +985,22 @@ class OpacityLayer extends ContainerLayer {
   /// (as described at [Layer]).
   int alpha;
 
+  /// Offset from parent in the parent's coordinate system.
+  Offset offset;
```

Comment:
should changing this or the alpha mark the layer as dirty?

Reply:
Yes, adding similar getters and setters to this Layer and other layers with mutable fields. Meanwhile, I'll also make as many Layer fields `final` as possible to minimize the change in this PR. If a Layer's fields become mutable in the future, we can add the related logic to just that Layer in that future PR. That should be much cleaner than a big change to every Layer in this PR.